pkg/apis/app/v1alpha1/kogitobuild_types.go

```diff
@@ -0,0 +1,236 @@
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Maven Mirror URL to be used during source-to-image builds (Local and Remote) to considerably increase build speed
+	// +optional
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Maven Mirror URL"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.x-descriptors="urn:alm:descriptor:com.tectonic.ui:label"
+	MavenMirrorURL string `json:"mavenMirrorURL,omitempty"`
+
+	// Image used to build the Kogito Service from source (Local and Remote).
+	// The operator will use the one provided by the Kogito Team based on the "Runtime" field.
+	// Example: Domain: quay.io, Namespace: kiegroup, Name: kogito-jvm-builder, Tag: latest.
+	// On OpenShift an ImageStream will be created in the current namespace pointing to the given image.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Build Image"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +optional
+	BuildImage Image `json:"buildImage,omitempty"`
+
+	// Image used as the base image for the final Kogito Runtime Service. This image only has the required packages to run the application.
+	// For example: quarkus based services will have only JVM installed, native services only the packages required by the OS.
+	// The operator will use the one provided by the Kogito Team based on the "Runtime" field.
+	// Example: Domain: quay.io, Namespace: kiegroup, Name: kogito-jvm-builder, Tag: latest.
+	// On OpenShift an ImageStream will be created in the current namespace pointing to the given image.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Base Image"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +optional
+	RuntimeImage Image `json:"runtimeImage,omitempty"`
+
+	// Set this field targeting the desired KogitoRuntime when this KogitoBuild instance has a different name than the KogitoRuntime.
+	// By default this KogitoBuild instance will generate a final image named after its own name (.metadata.name).
+	// On OpenShift, an ImageStream will be created causing a redeployment on any KogitoRuntime with the same name.
+	// On Kubernetes, the final image will be pushed to the KogitoRuntime deployment.
+	// If you have multiple KogitoBuild instances (let's say BinaryBuildType and Remote Source), you might need that both target the same KogitoRuntime.
+	// Both KogitoBuilds will update the same ImageStream or generate a final image to the same KogitoRuntime deployment.
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors.displayName="Base Image"
+	// +operator-sdk:gen-csv:customresourcedefinitions.specDescriptors=true
+	// +optional
+	TargetKogitoRuntime string `json:"targetKogitoRuntime,omitempty"`
```

Comment:
We can't add image streams to the `KogitoRuntime` since this is an OpenShift object, and we are all about supporting core Kubernetes API.

I got your point, we would have an array of possible images in the Runtime interface. From an UX perspective this seems awkward, a service having more than one image. Having the build targeting a service seems more natural IMO. I won't change that now, I'd rather see how this works for the community and hear more feedback from it out there. Think that the operator builds can be done on a separate namespace/environment, having a CI to promote the generated images. Having lot's of builds (building on different time frames) targeting the same image, would make things easier. Or, if that won't work for the user, they could just create a new build with no target runtime and promote the image retagging it. A list of images in the runtime is a no go to me at this moment.

> Because doing that way, I can update any runtime by mistake and with something that could completely unrelated ...

I don't follow. wdym?

Reply:
that can also be done the other way. I was myself a bit confused I think ...
Ok to let it like that and see the feedbacks